### PR TITLE
Replace DEPLOYMENT_OPTIONS to TEMPLATE_FEATURES

### PR DESCRIPTION
**Pull request by jonchenn**
Update to TEMPLATE_FEATURES to be able to take multiple feature tags.
The preferred format in the TEMPLATE_FEATURES string will be:
```
TEMPLATE_FEATURES="foundation|gke|cloudrun"
```

This will be updated in the documentation soon.

This is to prepare for easier deployment with other feature/stages in future.

### CODE REVIEW

File: setup/setup_all.sh [hunk 1]
```diff
@@ -135,14 +135,17 @@ link_billing_account
...
 fi
 
-if [[ "$MICROSERVICE_DEPLOYMENT_OPTION" ==  "cloudrun" ]]; then
+if [[ "$TEMPLATE_FEATURES" ==  "cloudrun" ]]; then
   printf "Deploying microservices to CloudRun...\n"
   deploy_microservices_to_cloudrun
   test_api_endpoints_cloudrun
+fi
+  deploy_microservices_to_cloudrun
+  test_api_endpoints_cloudrun
```
Comment (lavjain) [inline setup/setup_all.sh]: Most likely due to extra cut paste

File: e2e/run_e2e_test.sh [hunk 2]
```diff
@@ -116,12 +116,12 @@ install_dependencies
 setup_working_folder
 
 # Run setup_all script to deploy to GKE.
-export MICROSERVICE_DEPLOYMENT_OPTION="gke"
+export SOLUTIONS_TEMPLATE_FEATURES="gke"
```
Comment (lavjain) [inline e2e/run_e2e_test.sh]: TEMPLATE_FEATURES

File: {{cookiecutter.project_id}}/e2e/run_e2e_test.sh [hunk 3]
```diff
@@ -116,12 +116,12 @@ install_dependencies
 setup_working_folder
 
 # Run setup_all script to deploy to GKE.
-export MICROSERVICE_DEPLOYMENT_OPTION="gke"
+export SOLUTIONS_TEMPLATE_FEATURES="gke"
```
Comment (lavjain) [inline {{cookiecutter.project_id}}/e2e/run_e2e_test.sh]: TEMPLATE_FEATURES

File: setup/setup_all.sh (hunk 1 above)
Comment (jonchenn) [reply setup/setup_all.sh]: Removed in the new commit.

File: {{cookiecutter.project_id}}/e2e/run_e2e_test.sh (hunk 3 above)
Comment (jonchenn) [reply {{cookiecutter.project_id}}/e2e/run_e2e_test.sh]: Done

File: e2e/run_e2e_test.sh (hunk 2 above)
Comment (jonchenn) [reply e2e/run_e2e_test.sh]: Done

**Review by lavjain — APPROVED** [review]

**Pull request merged**